internal/dataplane/kong_client.go

```diff
@@ -451,14 +465,25 @@ func (c *KongClient) triggerKubernetesObjectReport(objs ...client.Object) {
 	// control-plane can update the Kubernetes object statuses for affected objs.
 	// this has to be done in a separate loop so that the filter is in place
 	// before the objects are enqueued, as the filter is used by the control-plane
-	for _, obj := range objs {
+	for _, obj := range reportedObjects {
+		c.logger.Infof("publishing event for object %s/%s/%s", obj.GetObjectKind().GroupVersionKind(), obj.GetNamespace(), obj.GetName())
 		c.kubernetesObjectStatusQueue.Publish(obj)
 	}
+
+	// TODO: there would be duplicate objects in translation failures, for example,
+	// it an object causes more than one translation failures, it will be published twice.
+	// so there may be requirement to de-duplicate the objects.
+	for _, translationFailure := range translationFailures {
+		for _, obj := range translationFailure.CausingObjects() {
+			c.logger.Infof("publishing event for object %s/%s/%s failed to translate", obj.GetObjectKind().GroupVersionKind(), obj.GetNamespace(), obj.GetName())
+			c.kubernetesObjectStatusQueue.Publish(obj)
+		}
+	}
```

Comment:
I think that we should be able to use objects stored in the `set` as they are already deduplicated there, right?

Reply:
`set` does not store objects in the current state of this PR. WDYT about the structure being stored in the `set`?

Reply:
created a new `set` to do the de-dup here.